### PR TITLE
[WIP] mgr/cephadm: provide container resources usage using cAdvisor

### PR DESCRIPTION
**Pull request by avanthakkar**
Fixes: https://tracker.ceph.com/issues/53399
Signed-off-by: Avan Thakkar <athakkar@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ def os_exit_noop(status: int) -> None:
 DEFAULT_NODE_EXPORTER_IMAGE = 'quay.io/prometheus/node-exporter:v0.18.1'
 DEFAULT_ALERT_MANAGER_IMAGE = 'quay.io/prometheus/alertmanager:v0.20.0'
 DEFAULT_GRAFANA_IMAGE = 'quay.io/ceph/ceph-grafana:6.7.4'
+DEFAULT_CADVISOR_IMAGE = 'gcr.io/cadvisor/cadvisor'
```
Comment (epuertat) [inline src/pybind/mgr/cephadm/module.py:98]: What tag?

**Review by epuertat — APPROVED** [review]
LGTM! Thanks @avanthakkar

**Comment by epuertat** [comment]
jenkins retest this please

**Comment by pcuzner** [comment]
@avanthakkar I tried cadvisor on the Pawsey cluster (4000 OSDs) and it generates a lot of data and demand on the Prometheus side as you scale. In fact so much so I only ended up using cadvisor on the mgr hosts. The issue manifested as data growth of the tsdb and  I/O impact - which since the default location is /var/lib started to impact the mon store too.

**Comment by epuertat** [comment]
> @avanthakkar I tried cadvisor on the Pawsey cluster (4000 OSDs) and it generates a lot of data and demand on the Prometheus side as you scale. In fact so much so I only ended up using cadvisor on the mgr hosts. The issue manifested as data growth of the tsdb and I/O impact - which since the default location is /var/lib started to impact the mon store too.

@pcuzner on my laptop cadvisor with the default set-up gave me more than 3,500 metrics. With the `-docker_only` and `-disable_root_cgroup_stats` flags (not sure if that work with podman too), it removes all the root cgroup stats and leaves just ~215 metrics.

If we also enable `cpuLoad` and `memory` metric sets that's just 97 metrics.

If we additionally set `--store_container_labels=false`, we reduce a lot the number of labels and reduce cardinality and payload of the `metrics` endpoint.

From this:

> container_cpu_load_average_10s{container_label_com_docker_compose_config_hash="c204013ba0e558c1907d9b19fbecb3c7ef008baf613a830f9d38ee2aaa4e8756",container_label_com_docker_compose_container_number="1",container_label_com_docker_compose_oneoff="False",container_label_com_docker_compose_project="dev",container_label_com_docker_compose_service="ceph",container_label_com_docker_compose_version="1.21.0",container_label_org_label_schema_build_date="20210915",container_label_org_label_schema_license="GPLv2",container_label_org_label_schema_name="CentOS Base Image",container_label_org_label_schema_schema_version="1.0",container_label_org_label_schema_vendor="CentOS",id="/system.slice/docker-84d140f254fb3d6fb3434eb014a104dd27fc3d627caaf688c9bd05e38b817ce9.scope",image="rhcsdashboard/ceph-rpm:master",name="ceph"} 0 1644434034206

... to this:
> container_cpu_load_average_10s{id="/system.slice/docker-84d140f254fb3d6fb3434eb014a104dd27fc3d627caaf688c9bd05e38b817ce9.scope",image="rhcsdashboard/ceph-rpm:master",name="ceph"} 0 1644433830274

**Pull request closed without merging**